rust/processor/src/processors/token_v2_processor.rs

```diff
@@ -186,6 +148,34 @@ async fn insert_collections_v2(
 }
 
 async fn insert_token_datas_v2(
+    conn: &mut MyDbConnection,
+    items_to_insert: Vec<TokenDataV2>,
+) -> Result<(), diesel::result::Error> {
+    match conn
+        .build_transaction()
+        .read_write()
+        .run::<_, Error, _>(|pg_conn| {
+            Box::pin(insert_token_datas_v2_impl(pg_conn, &items_to_insert))
+        })
+        .await
+    {
+        Ok(_) => Ok(()),
+        Err(_) => {
+            conn.build_transaction()
+                .read_write()
+                .run::<_, Error, _>(|pg_conn| {
+                    Box::pin(async {
+                        let token_datas_v2 = clean_data_for_db(items_to_insert, true);
```

Comment:
Not super related to this PR but why don't we just clean the data in the first place? When do we hit this block?

Reply:
this call is expensive. if you look at the function it goes through each value in the vec and checks if it's null . we want to avoid calling this if possible

Reply:
CPU is cheap, are we sure it's overall faster to avoid this function at the cost of maybe making two requests to the DB? If the error case is super rare then probably but otherwise, I/O is expenny.

Reply:
@bowenyang007 how often does this null values problem occur?

Reply:
Logs show that db write errors happen very infrequently ([link](https://cloud.us.humio.com/k8s/search?columns=%5B%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22%40timestamp%22%2C%22format%22%3A%22datetime%22%2C%22width%22%3A%22content%22%7D%2C%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22fields.message%22%2C%22format%22%3A%22text%22%2C%22width%22%3A587%7D%2C%7B%22type%22%3A%22field%22%2C%22fieldName%22%3A%22%40rawstring%22%2C%22format%22%3A%22logline%22%7D%5D&live=false&newestAtBottom=true&query=k8s.namespace%20%3D%20%22indexer-v2-*%22%0A%7C%20fields.message%20%3D%20%22Error%20running%20query%3A%20*%22&showOnlyFirstLine=false&start=30d&tz=America%2FNew_York&widgetType=list-view)). So I think this double call is fine in most cases.